x-pack/plugins/cases/public/components/actions/delete/use_delete_action.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import { EuiIcon, EuiTextColor, useEuiTheme } from '@elastic/eui';
+import { Case } from '../../../../common';
+import { useDeleteCases } from '../../../containers/use_delete_cases';
+
+import * as i18n from './translations';
+import { UseActionProps } from '../types';
+
+const getDeleteActionTitle = (totalCases: number): string =>
+  totalCases > 1 ? i18n.BULK_ACTION_DELETE_LABEL : i18n.DELETE_ACTION_LABEL;
+
+export const useDeleteAction = ({ onAction, onActionSuccess, isDisabled }: UseActionProps) => {
```

Comment:
I moved the logic to the hooks as you suggested. To avoid the consumers doing permissions checks, the hooks will return `canDelete` and `canUpdateStatus` respectively.